code/game/machinery/computer/buildandrepair.dm

```diff
@@ -113,9 +113,40 @@
 			if(P.tool_behaviour == TOOL_SCREWDRIVER)
 				P.play_tool_sound(src)
 				to_chat(user, "<span class='notice'>You connect the monitor.</span>")
-				var/obj/B = new circuit.build_path(loc)
-				B.setDir(dir)
-				transfer_fingerprints_to(B)
+
+				var/obj/machinery/computer/new_computer = new circuit.build_path(loc)
+				new_computer.setDir(dir)
+				transfer_fingerprints_to(new_computer)
+
+				if(istype(new_computer))
```

Comment:
What other type could it be? Could you set the variable to the base expected type and redefine it inside the istype check instead?

Reply:
Well, a circruit's build_path can basically be any type path. Circuits in general build both machines and computers. I can remove it if you feel it unnessary, or stack_trace it instead if it happens.

Reply:
just type it as the lowest level that can be reasonably be built, and then typecast it inside the type check block